camayoc/api.py

```diff
@@ -121,30 +126,88 @@ def __init__(self, response_handler=None, url=None):
         else:
             self.response_handler = response_handler
 
+        if authenticate:
+            self.login()
+            self._authenticated = True
+
+    def login(self):
+        """Login to the server to receive an authorization token."""
+        cfg = config.get_config().get('qcs', {})
+        server_username = cfg.get('username', 'admin')
+        server_password = cfg.get('password', 'pass')
+        login_request = self.request(
+            'POST',
+            urljoin(self.url, QCS_TOKEN_PATH),
+            json={
+                'username': server_username,
+                'password': server_password
+            }
+        )
+        self._authenticated = True
+        self.token = login_request.json()['token']
+        return login_request
+
+    def logout(self):
+        """Start sending unauthorized requests.
+
+        There is no API interaction that need occur to logout.
+        We simply must send unauthorized requests.
+        """
+        self._authenticated = False
+        self.token = None
```

Comment:
Sorry to change my mind again. So when I got to this point in the review I realized that the authenticated state can be controlled by the token, if it is None it is not authenticated else it is authenticated.

If you don't mind I think we can drop the _authenticated and just check if it is authenticated if the token is set. What do you think?

Reply:
Sounds good, that cuts down on some accounting